### PR TITLE
Add documentation generation

### PR DESCRIPTION
**Pull request by matthewfcarlson**
This uses portray (https://timothycrosley.github.io/portray/docs/quick_start/2.-cli/)
- pip install requirements.publisher.txt
- portray in_browser

To publish to github, run portray on_github_pages

### CODE REVIEW

**Comment by spbrogan** [comment]
1. Should we setup a pipeline to publish....either readthedocs.io or github.io
2. should we switch our other tools config over to the toml file?
3. I would have thought so of the docs themselves should be updated to include how to create, config, and publish the docs

Overall i really like this

**Comment by spbrogan** [comment]
Also i noticed the md files from the repo are not showing up

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+[[tool.portray.mkdocs.nav]]
+
+    [[tool.portray.mkdocs.nav.Features]]
+    "Ansi" = "docs/features/logging.ansi_handler.md"
+
+    [[tool.portray.mkdocs.nav.Features]]
+    "GetHostInfo" = "docs/features/utility_functions.GetHostInfo.md"
+
+    [[tool.portray.mkdocs.nav.Features]]
+    "Windows Firmware Policy" = "docs/features/windows_firmware_policy.md"
```
Comment (spbrogan) [inline pyproject.toml]: this seems like it needs to be auto generated

**Comment by corthon** [comment]
Is there still an example of the current output somewhere?

**Comment by matthewfcarlson** [comment]
https://matthewc.dev/edk2-pytool-library/

**Comment by matthewfcarlson** [comment]
Note: Look into release point

**Comment by matthewfcarlson** [comment]
We're going to use the version checked into pytool-extensions

**Pull request closed without merging**